File: pkg/templates/typescript/openai-computer-use/test.local.ts
import 'dotenv/config';
import { Kernel } from '@onkernel/sdk';
import { Agent } from './lib/agent';
import { KernelComputer } from './lib/kernel-computer';

/**
 * Local test script that creates a remote Kernel browser and runs the CUA agent.
 * No Kernel app deployment needed.
 *
 * Usage:
 *   KERNEL_API_KEY=... OPENAI_API_KEY=... npx tsx test.local.ts
 */

async function test(): Promise<void> {
  if (!process.env.KERNEL_API_KEY) throw new Error('KERNEL_API_KEY is not set');
  if (!process.env.OPENAI_API_KEY) throw new Error('OPENAI_API_KEY is not set');

  const client = new Kernel({ apiKey: process.env.KERNEL_API_KEY });
  const browser = await client.browsers.create({ timeout_seconds: 300 });
  console.log('> Browser session:', browser.session_id);
  console.log('> Live view:', browser.browser_live_view_url);

  const computer = new KernelComputer(client, browser.session_id);

  try {
    await computer.goto('https://duckduckgo.com');

    const agent = new Agent({
      model: 'computer-use-preview',
      computer,
      tools: [],
      acknowledge_safety_check_callback: (m: string): boolean => {
        console.log(`> safety check: ${m}`);
        return true;
      },
    });

    const logs = await agent.runFullTurn({
      messages: [
        {
          role: 'system',
          content: `- Current date and time: ${new Date().toISOString()} (${new Date().toLocaleDateString(
            'en-US',
            { weekday: 'long' },
          )})`,
        },
        {
          type: 'message',
          role: 'user',
          content: [
            {
              type: 'input_text',
              text: 'go to ebay.com and look up oberheim ob-x prices and give me a report',
            },
          ],
        },
      ],
      print_steps: true,
      debug: true,
      show_images: false,
    });
    console.dir(logs, { depth: null });
  } finally {
    await client.browsers.deleteByID(browser.session_id);
    console.log('> Browser session deleted');
  }
}

test();

File: pkg/templates/python/openai-computer-use/test_local.py
"""
Local test script that creates a remote Kernel browser and runs the CUA agent.
No Kernel app deployment needed.

Usage:
    KERNEL_API_KEY=... OPENAI_API_KEY=... uv run test_local.py
"""

import datetime
import os
import json

from dotenv import load_dotenv

load_dotenv(override=True)

from kernel import Kernel
from agent import Agent
from computers.kernel_computer import KernelComputer


def main():
    if not os.getenv("KERNEL_API_KEY"):
        raise ValueError("KERNEL_API_KEY is not set")
    if not os.getenv("OPENAI_API_KEY"):
        raise ValueError("OPENAI_API_KEY is not set")

    client = Kernel(api_key=os.getenv("KERNEL_API_KEY"))
    browser = client.browsers.create(timeout_seconds=300)
    print(f"> Browser session: {browser.session_id}")
    print(f"> Live view: {browser.browser_live_view_url}")

    computer = KernelComputer(client, browser.session_id)

    try:
        computer.goto("https://duckduckgo.com")

        items = [
            {
                "role": "system",
                "content": f"- Current date and time: {datetime.datetime.utcnow().isoformat()} ({datetime.datetime.utcnow().strftime('%A')})",
            },
            {
                "role": "user",
                "content": "go to ebay.com and look up oberheim ob-x prices and give me a report",
            },
        ]

        agent = Agent(
            computer=computer,
            tools=[],
            acknowledge_safety_check_callback=lambda message: (
                print(f"> safety check: {message}") or True
            ),
        )

        response_items = agent.run_full_turn(
            items,
            debug=True,
            show_images=False,
        )

        print(json.dumps(response_items, indent=2, default=str))
    finally:
        client.browsers.delete_by_id(browser.session_id)
        print("> Browser session deleted")


if __name__ == "__main__":
    main()
